cli/tests/unit/write_text_file_test.ts

```diff
@@ -1,4 +1,12 @@
 import { unitTest, assert, assertEquals } from "./test_util.ts";
+import { resolve, join } from "../../../std/path/mod.ts";
+
+const getResolvedUrl = (path: string): URL =>
+  new URL(
+    Deno.build.os === "windows"
+      ? "file:///" + resolve(path).replace(/\\/g, "/")
+      : "file://" + resolve(path)
+  );
```

Comment:
@actual-size The URL constructor does the backslash replacement for you, doesn't it?

```suggestion
function getResolvedUrl(path: string): URL {
  const url = new URL("file:///");
  url.pathname = resolve(path);
  return url;
}
```
This should handle every case correctly. Also see https://deno.land/manual/contributing/style_guide#top-level-functions-should-not-use-arrow-syntax.

Reply:
Note that you do need to manually handle percent decoding as well in the conversion to paths.

For reference the Node.js implementations can be seen here - https://github.com/nodejs/node/blob/master/lib/internal/url.js#L1290.

Node.js also specifically includes a check that no percent-encoded `/` is present to provide some vague security.

Reply:
@nayeemrmn Thanks for that, never thought to check if the constructor would do it for me. 
The exact snippet you've got there doesn't work, because the replacement only happens in the constructor, but I'll change my method to use the automatic replacement.

Reply:
@guybedford I think you've commmented on the wrong method here, but thanks for the link/advice about percent decoding! I'll start on working it into my implementation